### PR TITLE
Bind ElementArrayView<PyObject>

### PR DESCRIPTION
**Pull request by jl-wynen**
Needed to access the `values` of non-scalar Variables of Python objects.

### CODE REVIEW

File: python/element_array_view.cpp [hunk 1]
```diff
@@ -42,6 +44,32 @@ void declare_ElementArrayView(py::module &m, const std::string &suffix) {
...
+      .def(
+          "__setitem__",
+          [](ElementArrayView<T> &self, const scipp::index i,
+             const py::object &value) { return self[i].to_pybind() = value; },
+          py::return_value_policy::reference)
+      .def("__len__", &ElementArrayView<T>::size)
+      .def("__iter__", [](const ElementArrayView<T> &self) {
+        return py::make_iterator(self.begin(), self.end());
+      });
+}
```
Comment (SimonHeybrock) [inline python/element_array_view.cpp:71]: Quite some duplication, would it be briefer and more maintainable to use `if constexpr(std::is_same_v<T, PyObject>)` in the original template definition instead?

File: python/element_array_view.cpp [hunk 2]
```diff
@@ -57,6 +85,7 @@ void init_element_array_view(py::module &m) {
   declare_ElementArrayView<bucket<Variable>>(m, "bin_Variable");
   declare_ElementArrayView<bucket<DataArray>>(m, "bin_DataArray");
   declare_ElementArrayView<bucket<Dataset>>(m, "bin_Dataset");
+  declare_ElementArrayView<scipp::python::PyObject>(m, "PyObject");
```
Comment (SimonHeybrock) [inline python/element_array_view.cpp:88]: Also need the const version now (below)?

File: python/py_object.cpp [hunk 3]
```diff
@@ -36,4 +36,6 @@ bool PyObject::operator==(const PyObject &other) const {
   return to_pybind().equal(other.to_pybind());
 }
 
+std::string to_string(const PyObject &obj) { return py::str(obj.to_pybind()); }
```
Comment (SimonHeybrock) [inline python/py_object.cpp:39]: Should this acquire the GIL first? `py::str` might segfault otherwise?

Comment (jl-wynen) [reply python/py_object.cpp:39]: Good point

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**